Code Review Implementation

RICHYDOS opened the pull request:
### What is this change supposed to do?
- Show that I listened to the feedback and went through the resources provided.

### Summary of changes made. 
- Removed compiled files from the repository
- Abstracted the environment variables in the repository
- Changed the status codes and HTTP methods where necessary in the controller folder
- Removed Unnecessary comments from the codebase
- Renamed variable names and file names to be more descriptive.
-  Added a ".env.example" file.

Richydos2305 committed:
fix: Remove compiled files from the repository

Richydos2305 committed:
refactor: rename routes to fit standard practice

Richydos2305 committed:
refactor: abstract environment variables

Richydos2305 committed:
feature: add '.env example' file so that collaborators know an env file is needed

Richydos2305 committed:
feature: add authorisation to routes where 'CRUD' operations are performed on an existing user

Richydos2305 committed:
fix: Change status codes where necessary for the client

Richydos2305 committed:
fix: major- uninstalled and reinstalled 'sequelize-cli' to setup the database again with better naming conventions

Richydos2305 committed:
refactor: changed the look of the database.ts file

Richydos2305 committed:
fix: recreated the models and made 'variable name' changes where necessary in the controllers

Richydos2305 committed:
fix: modified the database file to fix an issue I had when staring the server

Richydos2305 committed:
refactor: renamed file names to be more descriptive

Richydos2305 committed:
refactor: removed unnecessary commits

Richydos2305 committed:
fix: rename model names in conroller files

norb-c reviewed (commented):
Nice work @RICHYDOS 
Please follow this [guide](https://www.coffeeclass.io/articles/commit-better-code-with-husky-prettier-eslint-lint-staged) to add eslint, prettier and husky to your code base. They help maintain coding standard and format your codes.

Richydos2305 committed:
chore: renamed 'app.ts' file

Richydos2305 committed:
fix: resolved errors when using the sequelize-cli package to create migrations

Richydos2305 committed:
fix: created the modified migration files

Richydos2305 committed:
Setup eslint, prettier, and husky

Richydos2305 committed:
Added more rules from the 'backend-assessment-template-main'

Richydos2305 committed:
test: test to see if husky is working

Richydos2305 committed:
chore: remove the line since husky is working

Richydos2305 committed:
feature: create loggers for the project using winston instead of console.log

Richydos2305 committed:
feature: add logger to the errorhandling middleware

Richydos2305 committed:
feature: add logger to the errorhandling middleware

had to make some tweaks in the code because of eslint and husky

Richydos2305 committed:
feature: add an error.log example file

Richydos2305 committed:
refactor: created an interface that extends the Request object instead of modifying the global Request object. Also added return types for all the functions

Richydos2305 committed:
fix: re-implement the models folder to have type annotations

had to forgo the sequelize-cli code because I struggled to convert the code to full typescript

Richydos2305 committed:
refactor: abstracted similar/frequently used queries into the DAL folder to make user controller file less clunky

Richydos2305 committed:
chore: add logger to remaining user controller endpoints

Richydos2305 committed:
feat: add hidden associations methods for certain models and add new interface for update methods

Richydos2305 committed:
refactor: abstracted similar/frequently used queries into the DAL folder to make remaining controller files less clunky

Richydos2305 committed:
refactor: implement the abstracted code from the dal folder

Richydos2305 committed:
refactor: uncomment the code to test the endpoints

Richydos2305 committed:
refactor: add type annotations for the controller parameter

Richydos2305 committed:
fix: fix issue regarding parameters in the errorhandler function

Richydos2305 committed:
chore: change statement in the catch block to send the error statement to the client

Richydos2305 committed:
refactor: make sure necessary variables or functions have their types properly defined
